ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineForkchoiceUpdated.java

```diff
@@ -60,101 +62,108 @@ public String getName() {
 
   @Override
   public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext) {
+    final Object requestId = requestContext.getRequest().getId();
 
     final EngineForkchoiceUpdatedParameter forkChoice =
         requestContext.getRequiredParameter(0, EngineForkchoiceUpdatedParameter.class);
     final Optional<EnginePayloadAttributesParameter> optionalPayloadAttributes =
         requestContext.getOptionalParameter(1, EnginePayloadAttributesParameter.class);
 
+    if (mergeContext.isSyncing()) {
+      return syncingResponse(requestId);
+    }
+
+    Optional<BlockHeader> newHead =
+        protocolContext.getBlockchain().getBlockHeader(forkChoice.getHeadBlockHash());
+
+    if (newHead.isEmpty()) {
+      Optional.ofNullable(forkChoice.getHeadBlockHash())
+          .filter(hash -> !hash.equals(Hash.ZERO))
+          .ifPresent(mergeCoordinator::getOrSyncHeaderByHash);
+
+      return syncingResponse(requestId);
+    }
+
+    LOG.info(
+        "Consensus fork-choice-update: head: {}, finalized: {}, safeBlockHash: {}",
+        forkChoice.getHeadBlockHash(),
+        forkChoice.getFinalizedBlockHash(),
+        forkChoice.getSafeBlockHash());
+
     Optional<Hash> maybeFinalizedHash =
         Optional.ofNullable(forkChoice.getFinalizedBlockHash())
             .filter(finalized -> !Hash.ZERO.equals(finalized));
 
+    if (!isPartOfCanonicalChain(forkChoice.getSafeBlockHash())
+        || (maybeFinalizedHash.isPresent() && !isPartOfCanonicalChain(maybeFinalizedHash.get()))) {
```

Comment:
one additional consideration - I asked a clarifying question about safeBlockHash here: 
https://discord.com/channels/595666850260713488/892088344438255616/975915304045649960

I think the spec leaves room for us to get 0x000..000 as safeBlockHash in the same way do for finalized, prior to the first finalized block.

Reply:
it looks like we should expect 0x00...00 as safeBlockHash prior to TTD being justified/finalized:

https://discord.com/channels/595666850260713488/892088344438255616/976024691766591508